### ISSUE TITLE
Can we capture common misspellings

### ISSUE THREAD

**Issue by tarunima** [tattle-made/Uli#311]
For the same slur we encounter multiple spellings/mis-spellings of slurs. This ticket is to find a way to capture the misspellings. It may be used to add to the slur list or deployed in Uli.

**Comment by duggalsu** [comment]
There are some performance trade-offs I expect to see with approximate slur matching
1. If we implement Levenstein distance in javascript within the plugin, I believe it will visibly slow down real-time slur replacement, especially on slower devices
2. The alternative is to create a list offline and add it to our plugin slur list. However, we will have to test the performance impact if this list gets too large. Maybe limit the edit distance to 1 insert/delete/substitute/transpose and test.

**Comment by duggalsu** [comment]
Ideally, what we really want to achieve is finding words that sound the same, even if they have different spellings. This is addressed through Phonetic Algorithms.

- https://en.wikipedia.org/wiki/Phonetic_algorithm
```
Most phonetic algorithms were developed for English and are not useful for indexing words in other languages.
```

There have been some attempts to build Phonetic algorithms for Indic languages.
1. https://thottingal.in/blog/2009/07/26/indicsoundex/
2. https://www.researchgate.net/publication/316891126_Need_for_Customized_Soundex_based_Algorithm_on_Indian_Names_for_Phonetic_Matching
3. https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3579322

I propose using and building on the method from the third link above.

Proposed algorithm - 
1. Transliterate Indic language slurs to Roman script to handle multiple languages.
2. Create custom groupings for slurs based on the Common key method. This will require phonetic inputs from native language speakers.
3. Use the edit distance method to capture misspellings to increase accuracy.

**Comment by tarunima** [comment]
The reason to define this task and #310 as distinct tasks was to allow for some creative solutions for approximate matching. For example, we could think of this task as generating a list of possible misspellings from a root (not necessarily detecting it as a misspelling of a root word), and then using regex on that list. 
We can give the phonetic algorithm a try, but I am not sure the misspellings, for all slurs, if said out a loud will sound phonetically similar. I am thinking of words like Ola-o-uber and Ola Hi Uber Hai. Or say where people put dollar signs in place of certain alphabets.

**Comment by duggalsu** [comment]
Addressed by PR #327
